src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -983,6 +1031,103 @@ get_vbnv(const axlf* top)
   return {vbnv, strnlen(vbnv, vbnv_length)};
 }
 
+std::string
+get_device_name(const char* xmlFile,size_t xmlFileSize)   
+{
+  std::string fpgaDevice="";
+  pt::ptree xml_project;
+  std::string sXmlFile;
+  sXmlFile.assign(xmlFile, xmlFileSize);
+  std::stringstream xml_stream;
+  xml_stream << sXmlFile;
+  pt::read_xml(xml_stream, xml_project);
+	
+  // iterate devices
+  int count = 0;
+  for (auto& xml_device : xml_project.get_child("project.platform"))
+  {
+    if (xml_device.first != "device")
+      continue;
+
+    fpgaDevice = xml_device.second.get<std::string>("<xmlattr>.fpgaDevice");
+
+    if (++count > 1)
+    {
+      //Give error and return from here
+      return "";
+ 
+    }
+  }
+    
+  return fpgaDevice;
+	
+}
+
+bool
+check_if_multiple_platform_exists(const char* xml_data, size_t xml_size)
+{
+  pt::ptree xml_project;
+  std::stringstream xml_stream;
+  xml_stream.write(xml_data,xml_size);
+  pt::read_xml(xml_stream,xml_project);
+  int count = 0;
+
+  for (auto& xml_platform : xml_project.get_child("project"))
+  {
+    if (xml_platform.first != "platform")
+      continue;
+    if (++count > 1)
+    {
+      //Give error and return from here
+      return false;
+    }
+  }
+  return true;
+}
+
+bool
+check_if_multiple_core_exists(const char* xml_data, size_t xml_size)
+{
+  pt::ptree xml_project;
+  std::stringstream xml_stream;
+  xml_stream.write(xml_data,xml_size);
+  pt::read_xml(xml_stream,xml_project);
+  int count = 0;
+  for (auto& xml_core : xml_project.get_child("project.platform.device"))
+  {
+    if (xml_core.first != "core")
+      continue;
+    if (++count > 1)
+    {
+      //Give error and return from here
+      return false;
+    }
+  }
+  return true;
+}
+
+std::map<std::string,uint64_t>
+get_address_range_map(const char* xml_data, size_t xml_size)
+{
```

Comment:
This is a kernel property which is accessible via core/common/api/xclbin_int.h:get_properties(kernel).